tests/embind/test_val.cpp

```diff
@@ -570,19 +571,41 @@ int main()
   ensure_js("test_val_throw_(new TypeError('message'))");
   
   // this test should probably go elsewhere as it is not a member of val
-  test("template<typename T> std::vector<T> vecFromJSArray(val v)");
+  test("template<typename T> std::vector<T> vecFromJSArray(const val& v)");
```

Comment:
I don't know the embind code or tests well, but it would be better to add new tests rather than modify existing ones - to make review easier, and to avoid the risk of losing some test coverage. Unless this is actually a necessary change?

Reply:
I can split it if you want, I just thought that there was no need to go through `EM_ASM` for almost the same setup code

Reply:
Let's see what @jgravelle-google thinks - maybe I don't know embind enough to tell if these test changes are obviously ok or not.

Reply:
Ping @jgravelle-google to know if the changes are ok or not